.build/microsoft-azure/build-vm.sh

```diff
@@ -0,0 +1,152 @@
+
+# az feature show --namespace Microsoft.VirtualMachineImages --name VirtualMachineTemplatePreview | grep state
+
+# az feature show --namespace Microsoft.KeyVault --name VirtualMachineTemplatePreview
+
+# # wait until it says registered
+
+# # check you are registered for the providers
+
+# az provider show -n Microsoft.Compute -o json | grep registrationState
+# az provider show -n Microsoft.Storage -o json | grep registrationState
+# az provider show -n Microsoft.KeyVault -o json | grep registrationState
+# az provider show -n Microsoft.VirtualMachineImages -o json | grep registrationState
+# az provider show -n Microsoft.Network -o json | grep registrationState
+
+#
+# Set Permissions & Create Resource Group for Image Builder Images
+#
+
+# set your environment variables here!!!!
+
+# destination image resource group
+export imageResourceGroup=nextcloud-aio-build
+
+# location (see possible locations in main docs)
+export location=westeurope
+
+# your subscription
+# get the current subID : 'az account show | grep id'
+export subscriptionID=$(az account show | grep id | tr -d '",' | cut -c7-)
+
+# name of the image to be created
+export imageName=nextcloud-aio-image
+
+# image distribution metadata reference name
+export runOutputName=nextcloudAioImg01ro
+
+# create resource group
+# az group create -n $imageResourceGroup -l $location
+
+
+#
+# Create User-Assigned Managed Identity and Grant Permissions
+#
+
+# create user assigned identity for image builder to access the storage account where the script is located
+export identityName=nextcloud-aio-$(date +'%s')
+az identity create -g $imageResourceGroup -n $identityName
+
+# get identity id
+export imgBuilderCliId=$(az identity show -g $imageResourceGroup -n $identityName | grep "clientId" | cut -c16- | tr -d '",')
+
+# get the user identity URI, needed for the template
+export imgBuilderId=/subscriptions/$subscriptionID/resourcegroups/$imageResourceGroup/providers/Microsoft.ManagedIdentity/userAssignedIdentities/$identityName
+
+# download preconfigured role definition example
+rm -f aibRoleImageCreation.json
+curl https://raw.githubusercontent.com/Azure/azvmimagebuilder/main/solutions/12_Creating_AIB_Security_Roles/aibRoleImageCreation.json -o aibRoleImageCreation.json
+
+export imageRoleDefName="Azure Image Builder Image Def"$(date +'%s')
+
+# update the definition
+sed -i -e "s/<subscriptionID>/$subscriptionID/g" aibRoleImageCreation.json
+sed -i -e "s/<rgName>/$imageResourceGroup/g" aibRoleImageCreation.json
+sed -i -e "s/Azure Image Builder Service Image Creation Role/$imageRoleDefName/g" aibRoleImageCreation.json
+
+# create role definitions
+az role definition create --role-definition ./aibRoleImageCreation.json
+
+# grant role definition to the user assigned identity
+az role assignment create \
+    --assignee $imgBuilderCliId \
+    --role $imageRoleDefName \
+    --scope /subscriptions/$subscriptionID/resourceGroups/$imageResourceGroup
```

Comment:
I am currently here but have run into https://github.com/MicrosoftDocs/azure-docs/issues/75826 so not sure how to proceed...